### PR TITLE
operation logs: implement operation history

### PR DESCRIPTION
**Pull request by Garfield-fr**
* Implements operation history on documents, holdings and items.
* Updates the holdings route definition to use a 'long mode' editor for
  holdings.
* Fixes the inherited call number if record contain metadata
  informations.

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

## Dependencies

My PR depends on `rero-ils#1637`'s PR(s):

* https://github.com/rero/rero-ils/pull/1637

## How to test?

New holdings fields:
* patterns.language
* issue_binding
* acquisition_status
* acquisition_method
* acquisition_expected_end_date
* general_retention_policy
* completeness
* composite_copy_report

Holdings masking:
* new field holdings._masked
* masked holdings are not displayed in the public interface (feature to be complete the next sprint)

Holdings Editor:

* enable long mode
* `patterns` field is optional
* at the creation of new holding, only these fields are added by default:
** holdings.location
** holdings.circulation_category
** holdings.call_number
** holdings.EnumerationAnyChronology
** holdings.vendor
** holdings._masked

Document index:
The following holdings fields are indexed in the document record:
* holdings.call_number
* holdings.second_call_number
* holdings.index
* holdings.enumerationAndChronology
* holdings.supplementaryContent
* holdings.notes

Operation log:
* Capture create and update operations for items/holdings/documents
* Users to display operation logs for items/holdings/documents in frontend
* Check button "Operation history" on documents, holdings, items

Views:
* Show new fields on holdings detail view

Fix:
* fix inherit call number for item in item detailed view

UI: Since the translation is not yet complete the  `{{ counter }} hidden operation logs` and `{{ counter }} hidden operation log`, the text does not display the counter

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

**Comment by pronguen** [comment]
Some details to improve if we can:
- [ ] Display exactly the same holdings fields in the document detailed view than in the top section of the holdings detailed view. Currently, the missing issues for ex. should be displayed in the top section of the holdings detailed view (see image).
image
- [ ] Group in the holdings editor the 4 fields:
   - Collection disponible
   - Contenu supplémentaire
   - Index
   - Fascicules manquants
- [ ] Put in the holdings editor the field `vendor` just below `serial_pattern`

**Comment by JoelleDosimont** [comment]
Some remarks from my tests : 
- Masked field should be required (don't allow to delete).
- When a holding is masked, I should see the masked information in the professional interface (the only way to see this is currently to edit the holding).
- There is 2 "delete" icons for the Vendor.
- The "expected acquisition end date" has no label, which may be confusing.
- Later, the order of the fields can be sort differently (maybe to group certain type of information together : collection, acquisition, reclamation, prediction, binding, …).

However, those are not blocking issues ;-)

File: projects/admin/src/app/api/operation-logs-api.service.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+   * Get Operation logs by resource type and resource pid
+   * @param resourceType - string
+   * @param resourcePid - string
+   * @param action - string
+   * @param page - number
+   * @param itemPerPage - number
+   * @param sort - string
+   * @return Observable
+   */
+  getByResourceTypeAndPid(
```
Comment (zannkukai) [inline projects/admin/src/app/api/operation-logs-api.service.ts:43]: Just to find something to say : Why not simply named "getLogs" ?

File: projects/admin/src/app/record/detail-view/organisation-detail-view/organisation-detail-view.component.ts [hunk 2]
```diff
@@ -18,33 +18,43 @@ import { Component } from '@angular/core';
...
     return this._userService.user.isSystemLibrarian;
-
   }
+
+  /** Constructor
+   * @param _userService : UserService
+   */
+  constructor(
+    private _userService: UserService,
+    private _operationLogsService: OperationLogsService
```
Comment (zannkukai) [inline projects/admin/src/app/record/detail-view/organisation-detail-view/organisation-detail-view.component.ts:58]: missing docstring

File: projects/admin/src/app/record/detail-view/vendor-detail-view/vendor-detail-view.component.html [hunk 3]
```diff
@@ -81,6 +81,20 @@ <h1 class="mb-3">{{ record.metadata.name }}</h1>
...
           </tab>
         </ng-container>
+        <!-- OPERATION LOGS TAB -->
+        <tab *ngIf="isTabOperationLog" id="vendor-operation-log-tab" tabOrder="6" (selectTab)="showOperationLogs = true">
+          <ng-template tabHeading>
+            <i class="fa fa-list-ul mr-1"></i><span translate>Operation logs</span>
+          </ng-template>
+          <div class="mt-3">
+            <admin-operation-logs
+              resourceType="documents"
```
Comment (zannkukai) [inline projects/admin/src/app/record/detail-view/vendor-detail-view/vendor-detail-view.component.html:91]: vendors instead documents

**Review by zannkukai — APPROVED** [review]

File: projects/admin/src/app/api/operation-logs-api.service.ts (hunk 1 above)
Comment (Garfield-fr) [reply projects/admin/src/app/api/operation-logs-api.service.ts:43]: done

File: projects/admin/src/app/record/detail-view/organisation-detail-view/organisation-detail-view.component.ts (hunk 2 above)
Comment (Garfield-fr) [reply projects/admin/src/app/record/detail-view/organisation-detail-view/organisation-detail-view.component.ts:58]: done

**Review by iGormilhit — CHANGES_REQUESTED** [review]
Commit message proposition:

```
operation logs: implement operation history

* Implements operation history on documents, holdings and items.
* Updates the holdings route definition to use a 'long mode' editor for
  holdings.
* Fixes the inherited call number if record contain metadata
  informations.
```

File: projects/admin/src/app/record/detail-view/vendor-detail-view/vendor-detail-view.component.html (hunk 3 above)
Comment (Garfield-fr) [reply projects/admin/src/app/record/detail-view/vendor-detail-view/vendor-detail-view.component.html:91]: done

**Review by iGormilhit — APPROVED** [review]

File: projects/admin/src/app/record/operation-logs/operation-logs-dialog/operation-logs-dialog.component.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import { BsModalRef, BsModalService } from 'ngx-bootstrap/modal';
+import { OperationLogsComponent } from '../operation-logs.component';
+
+@Component({
+  selector: 'admin-operation-logs-dialog',
+  templateUrl: './operation-logs-dialog.component.html'
+})
+export class OperationLogsDialogComponent {
+
+  @Input() resourceType: string;
```
Comment (lauren-d) [inline projects/admin/src/app/record/operation-logs/operation-logs-dialog/operation-logs-dialog.component.ts:27]: Missing docstring

**Review by lauren-d — APPROVED** [review]

File: projects/admin/src/app/record/operation-logs/operation-logs-dialog/operation-logs-dialog.component.ts (hunk 4 above)
Comment (Garfield-fr) [reply projects/admin/src/app/record/operation-logs/operation-logs-dialog/operation-logs-dialog.component.ts:28]: done

**Pull request merged**